clang_format/__init__.py

```diff
@@ -3,11 +3,11 @@
 import sys
 
 
-def _get_executable(name):
+def get_executable(name):
     return os.path.join(os.path.dirname(__file__), "data", "bin", name)
```

Comment:
Currently this method doesn't actually return the correct path, since on Windows it leaves off the `.exe` extension. It happens to work with the current use case because if you ask Windows to *execute* a file it can automatically add the `.exe` extension, but if you do things like `Path(clang_format._get_executable('clang-format')).exists()` you'll get `False`, since the path returned isn't actually the path of the file. 

If we're making this a public method we should make it actually return the correct path. We can do this by calling `sysconfig.get_config_var("EXE")`, which returns the appropriate executable suffix for the current platform (".exe" on Windows and an empty string on other platforms). 

```suggestion
    return os.path.join(os.path.dirname(__file__), "data", "bin", name + sysconfig.get_config_var("EXE")) 
```
(Also need to add `import sysconfig`)

For an example, see [ruff's source code](https://github.com/astral-sh/ruff/blob/bc0a5aa4098a5b4d03365a22384b0307a124391c/python/ruff/__main__.py#L11). For the value of the `sysconfig` variables on various platforms, see https://gist.github.com/flaviut/0e3f87d28583d713e7412c65a4ebce7f

Reply:
Thanks for fixing :)